lib/saluki-components/src/sources/checks/mod.rs

```diff
@@ -101,8 +106,11 @@ impl Source for ChecksSource {
 
         info!("Checks source started.");
 
+        let (check_metrics_tx, mut check_metrics_rx) = mpsc::channel(10_000_000);
```

Comment:
Bounded channels: good.
Bounded at 10 million items: no no no no no no. :P

You can almost always start with something closer to like 64 - 128 for a bounded channel.